zallet/src/commands/migrate_zcashd_wallet.rs

```diff
@@ -314,21 +259,26 @@ impl MigrateZcashdWalletCmd {
         // range. We don't have a good source of individual per-account birthday information at
         // this point; once we've imported all of the transaction data into the wallet then we'll
         // be able to choose per-account birthdays without difficulty.
-        let wallet_birthday = Self::get_birthday(
-            &chain_subscriber,
-            // Fall back to the chain tip height, and then Sapling activation as a last resort. If
-            // we have a birthday height, max() that with sapling activation; that will be the
-            // minimum possible wallet birthday that is relevant to future recovery scenarios.
-            tx_heights
-                .values()
-                .map(|(h, _)| h)
-                .min()
-                .copied()
-                .or(chain_tip)
-                .map_or(sapling_activation, |h| std::cmp::max(h, sapling_activation)),
-            chain_tip,
-        )
-        .await?;
+        //
+        // Fall back to the chain tip height, and then Sapling activation as a last resort. If
+        // we have a birthday height, `max()` that with Sapling activation; that will be the
+        // minimum possible wallet birthday that is relevant to future recovery scenarios.
+        let birthday_height = tx_heights
+            .values()
+            .flat_map(|(h, _)| h)
+            .min()
+            .copied()
+            .or(chain_tip)
+            .map_or(sapling_activation, |h| std::cmp::max(h, sapling_activation));
+
+        // Fetch the tree state corresponding to the last block prior to the wallet's
+        // birthday height.
+        let chain_state = chain_view
+            .tree_state_as_of(birthday_height.saturating_sub(1))
+            .await?
+            .expect("birthday_height was sourced from chain_view");
```

Comment:
It seems odd to me to `.expect` in a function that returns an error enum. Is there a specific reason for this?